pkg/docker/config/config.go

```diff
@@ -606,6 +636,45 @@ func findAuthentication(registry, path string, legacyFormat bool) (types.DockerA
 	return types.DockerAuthConfig{}, nil
 }
 
+func findAuthenticationForRef(auths dockerConfigFile, ref reference.Named) (types.DockerAuthConfig, error) {
+	for _, registry := range pathsForRef(ref) {
```

Comment:
The iteration variable is not a `registry`. (Also maybe `authKeysForRef`, not sure about that one.)

Reply:
Changed the function name with the next push.